query_test.go

```diff
@@ -1952,7 +1946,7 @@ func TestSparseQueryWriteNullable(t *testing.T) {
 
 	// Create write buffers
 	bufferA1 := []int32{1, 2, 3}
-	bufferA1Validity := []uint8{1, 1, 0}
+	bufferA1Validity := []uint8{1, 1, 0, 0}
```

Comment:
Why is there 4 validity values here but the data buffer only has 3 values?

Reply:
Looks like the read buffer had an extra value leading to the mismatch, I've updated bufferA1 to have a fourth value to reflect readBufferA1Validity

Reply:
I don't think 4 values is correct, there is only 3 values set for dim1. Why is this not erroring out with a mismatch between the number of cells being written?

Reply:
That I'm not sure. It passes if I set them all back to 3 values

Reply:
It seems to me like this was not errored in the past and the test was wrong with the value of 4. New TileDB caught this an did error. So it was a mistake in the test (and in me changing the others to 4)